app/src/main/java/com/example/cinemapp/ui/main/ActorDetailsFragment.kt

```diff
@@ -0,0 +1,140 @@
+
+        observeFlowSafely(viewModel.state) {
+            it.details?.let { details ->
+                setupView(details)
+                binding.clContent.visibility = View.VISIBLE
+                binding.cpiLoading.visibility = View.GONE
+            }
+            val adapter = ArrayAdapter(
+                binding.root.context,
+                org.koin.android.R.layout.support_simple_spinner_dropdown_item,
+                viewModel.state.value.creditYears.map { year -> year?.toString() ?: "Unannounced" }
+            )
+            binding.spinnerYear.adapter = adapter
+        }
+        observeFlowSafely(viewModel.stateCredits) {
+            creditAdapter.setCredits(it.credits)
+        }
+    }
+
+    private fun setupView(details: PersonDetails) {
+        with(binding) {
+            toolbar.setNavigationIcon(R.drawable.vic_arrow_back)
+            toolbar.setNavigationOnClickListener {
+                requireActivity().onBackPressedDispatcher.onBackPressed()
+            }
+            tvName.text = details.name
+            val ageDisplay = details.birthday.ageAndRangeUntil(details.deathday)
+            tvAge.text = ageDisplay
+            tvGender.text = details.gender
+
+            setExpandableTextView(
+                text = details.biography,
+                phraseColor = requireContext().getColor(R.color.md_theme_tertiary),
+                maxChars = BIOGRAPHY_MAX_CHARACTERS,
+                textView = tvBiography
+            ) {
+                tvBiography.text = details.biography
+            }
+            loadImage(
+                details.profilePath,
+                ivPicture,
+                root.context,
+                R.drawable.ic_placeholder_person
+            )
+        }
+    }
+
+    private fun setupAdapter() {
+        with(binding) {
+            val adapter = ArrayAdapter(
+                root.context,
+                org.koin.android.R.layout.support_simple_spinner_dropdown_item,
+                viewModel.state.value.creditYears.map { year -> year?.toString() ?: "Unannounced" })
+
+            spinnerYear.adapter = adapter
+            spinnerYear.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+                override fun onItemSelected(
+                    parent: AdapterView<*>?,
+                    view: View?,
+                    position: Int,
+                    id: Long
+                ) {
+                    viewModel.getCreditsFromYear(
+                        args.personId,
+                        viewModel.state.value.creditYears[position]
+                    )
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {
+
+                }
+
+            }
+            rvCredits.adapter = creditAdapter
+            rvCredits.layoutManager =
+                LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+            observeFlowSafely(creditAdapter.onCardClick) {
```

Comment:
This adapter observations should be moved to a method called `observeClickEvents()`. That method should be called in `onCreateView`, instead of inside of `setupAdapter`